src/_pytest/config/__init__.py

```diff
@@ -1078,12 +1085,33 @@ def _checkversion(self):
                     )
                 )
 
-    def _validatekeys(self):
+    def _validate_keys(self) -> None:
         for key in sorted(self._get_unknown_ini_keys()):
-            message = "Unknown config ini key: {}\n".format(key)
-            if self.known_args_namespace.strict_config:
-                fail(message, pytrace=False)
-            sys.stderr.write("WARNING: {}".format(message))
+            self._emit_warning_or_fail("Unknown config ini key: {}\n".format(key))
+
+    def _validate_plugins(self) -> None:
+        required_plugins = sorted(self.getini("required_plugins"))
+        if not required_plugins:
+            return
+
+        plugin_info = self.pluginmanager.list_plugin_distinfo()
```

Comment:
Any particular reason to only include setuptools plugins here?

Reply:
This isn’t just setup tools it is all plugins, but with the proper distribution info. Correct me if I’m wrong @nicoddemus

Reply:
Resolving as above. Please reopen if I’m wrong

Reply:
`list_plugin_distinfo` only lists setuptools plugins, at least this is what I understand from the pluggy code. That might be what we want here, not sure myself, but unresolving so can figure it out.

Reply:
Good point @bluetech.

Should we consider somehow plugins specified via `-p` and `PYTEST_PLUGINS`?

Reply:
Do we want to add support for -p and PYTEST_PLUGINS in a follow up PR or does that feel like we’re submitting a half complete feature @nicoddemus @bluetech

Reply:
I haven't checked what each means, but just from perusing the `PlugingManager` code, `list_name_plugin()` seems good?

I guess it depends on why the code uses `dist.project_name` instead of the name the plugin was registered with? (Sorry if was discussed before)

Reply:
This was discussed here: https://github.com/pytest-dev/pytest/pull/7330#discussion_r437135961

Reply:
From reading the help, I think we should defer `-p` if we add it at all. It sounds like a user is intentionally fiddling around with the run
```
-p name               early-load given plugin module name or entry point
                            (multi-allowed).
```

The same goes for the environment variable `PYTEST_PLUGINS`. I'm not opposed to adding them in, but to me it really sounds like they're a different use case than this and would warrant their own PR

Reply:
Agree, not sure this makes sense when used with `-p` or `PYTEST_PLUGINS`.